src/test/groovy/org/prebid/server/functional/tests/AuctionSpec.groovy

```diff
@@ -431,4 +440,107 @@ class AuctionSpec extends BaseSpec {
         where:
         bidderName << [GENERIC, GENERIC_CAMEL_CASE]
     }
+
+    def "PBS should set device.ext.cdep from header when cookieDeprecation and Deprecation header is specified"() {
+        given: "Default basic BidRequest with generic bidder"
+        def bidRequest = BidRequest.defaultBidRequest
+
+        and: "Account in the DB"
+        def privacySandbox = PrivacySandbox.defaultPrivacySandbox
+        def accountAuctionConfig = new AccountAuctionConfig(privacySandbox: privacySandbox)
+        def accountConfig = new AccountConfig(status: ACTIVE, auction: accountAuctionConfig)
+        def account = new Account(uuid: bidRequest.accountId, config: accountConfig)
+        accountDao.save(account)
+
+        and: "Sec-Cookie-Deprecation header"
+        def secCookieDeprecation = [(COOKIE_DEPRECATION_HEADER): PBSUtils.randomString]
+
+        when: "PBS processes auction request with header"
+        defaultPbsService.sendAuctionRequest(bidRequest, secCookieDeprecation)
+
+        then: "BidResponse should have device.ext.cdep from sec-cookie-deprecation header"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.device.ext.cdep == secCookieDeprecation[COOKIE_DEPRECATION_HEADER]
+    }
+
+    def "PBS shouldn't set device.ext.cdep from header when cookieDeprecation is #privacySandbox and Deprecation header is #deprecationHeader"() {
+        given: "Default basic BidRequest with generic bidder"
+        def bidRequest = BidRequest.defaultBidRequest
+
+        and: "Account in the DB"
+        def accountAuctionConfig = new AccountAuctionConfig(privacySandbox: privacySandbox)
+        def accountConfig = new AccountConfig(status: ACTIVE, auction: accountAuctionConfig)
+        def account = new Account(uuid: bidRequest.accountId, config: accountConfig)
+        accountDao.save(account)
+
+        and: "Sec-Cookie-Deprecation header"
+        def secCookieDeprecation = [(COOKIE_DEPRECATION_HEADER): PBSUtils.randomString]
```

Comment:
Don't really need set sec-cookie-deprecation header when privacy sandbox is null or false